### PR TITLE
Session Manager init

### PR DESCRIPTION
**Pull request by JoshuaMitchell**

### CODE REVIEW

**Comment by JoshuaMitchell** [comment]
@karpet lots to unpack here (sorry this pr kind of got away from me).  Hybrid approach to your full session manager vs straight controller logic.  This makes some sense to me, but let me know if you have concerns.

File: app/main/tests/test_index.py [hunk 1]
```diff
@@ -54,3 +59,34 @@ def test_update_name_step_0_session_exists_already(app,session,client):
...
+
+
+def test_registered_voter_input_returns_redirect_step_AB_1(app, session, client):
+    """
+    An already registered voter returns a redirect to ab 1
+    """
+    form_payload = {
+            "name_first": "Jake",
+            "name_last": "Lowen",
+            "dob":"02/07/1979",
```
Comment (karpet) [inline app/main/tests/test_index.py]: let's leave Jake's PII out of the tests? how about we use the same one as for the ksmyvoteinfo tests?

https://github.com/BlueprintKansas/ksmyvoteinfo-py/blob/master/tests/test_ksmyvoteinfo.py#L15

File: app/services/session_manager.py [hunk 2]
```diff
@@ -1,22 +1,74 @@
...
-                    return self.current_step.prev_step.endpoint
+        # For Step 0 when no previous step exists
+        if not self.prev_step:
+            if self.current_step.is_complete:
+                return self.next_step.endpoint
             else:
-                if not self.registrant.registration_value.get(req):
-                    return self.current_step.prev_step.endpoint
-        return self.current_step.next_step
+                # could use some error message logic here... but a user shouldn't ever get here (famous last words)
```
Comment (karpet) [inline app/services/session_manager.py]: if the user should not get here, how about we raise an exception instead?

File: app/services/session_manager.py [hunk 3]
```diff
@@ -1,22 +1,74 @@
...
+
+        # if the step has been completed move on
+        if self.current_step.is_complete:
+            return self.next_step.endpoint
+
+        #default to returning current step
+        return self.current_step.endpoint
+
+
+    def check_value_of_requirement(req):
```
Comment (karpet) [inline app/services/session_manager.py]: I admit this seems to break encapsulation to have the SessionManager peering into the Registrant object so much. Could we instead put this logic into `Registrant` instead?
```python
return registrant.has_value_for_req(req)
```

File: app/main/tests/test_index.py (hunk 1 above)
Comment (JoshuaMitchell) [reply app/main/tests/test_index.py]: Good call.  didn't see those tests there on the ksmyvoteinfo

File: app/services/session_manager.py (hunk 2 above)
Comment (JoshuaMitchell) [reply app/services/session_manager.py]: Agreed.  I'm going to leave as is until I get an error handler in place.

**Pull request merged**